compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -586,7 +603,7 @@ pub fn compile_declarative_macro(
         transparency,
         lhses,
         rhses,
-        valid,
+        validity: result,
```

Comment:
I wonder if we should agree on a single naming scheme here. Currently you use `valid`, `validity` and `result`. Maybe use `validity` everywhere (only in the places you had to modify)?. Not required tho

Reply:
I was unhappy with all my tries but I found `emitted_err` which is kinda what the variable represents; also replaced the `mut` ones with `Option<ErrorGuaranteed>` instead of `Result<(), ErrorGuaranteed>` (which I use for function as they may want to return something someday).